include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -125,46 +121,36 @@ namespace seqan3
  * The SAM header (if present) is read/written once in the beginning before the
  * first record is read/written.
  */
-struct format_sam
+struct format_sam : private detail::format_sam_base
```

Comment:
OK, I understand the private inheritance now and it's the right thing to do :+1: 

But still make classes of the types, please!